Sources/Plasma/FeatureLib/pfPython/cyPythonInterface.cpp

```diff
@@ -1749,44 +1749,28 @@ PyObject* PythonInterface::RunFunction(PyObject* module, const char* name, PyObj
     return result;
 }
 
-PyObject* PythonInterface::ParseArgs(const char* args)
-{
-    PyObject* result = nullptr;
-    PyObject* scope = PyDict_New();
-    if (scope) 
-    {
-        //- Py_eval_input makes this function accept only single expresion (not statement)
-        //- When using empty scope, functions and classes like 'file' or '__import__' are not visible
-        result = PyRun_String(args, Py_eval_input, scope, nullptr);
-        Py_DECREF(scope);
-    }
-   
-    return result;
-}
-
-bool PythonInterface::RunFunctionSafe(const char* module, const char* function, const char* args) 
+bool PythonInterface::RunFunctionStringArg(const char* module, const char* name, const ST::string& arg)
 {
     PyObject* moduleObj = ImportModule(module);
     bool result = false;
-    if (moduleObj) 
-    {
-        PyObject* argsObj = ParseArgs(args);
-        if (argsObj) 
-        {
-            PyObject* callResult = RunFunction(moduleObj, function, argsObj);
-            if (callResult) 
-            {
-                result = true;
-                Py_DECREF(callResult);
+    if (moduleObj) {
+        PyObject* argObj = PyUnicode_FromSTString(arg);
+        if (argObj) {
+            PyObject* argsObj = PyTuple_Pack(1, argObj);
+            if (argsObj) {
+                PyObject* callResult = RunFunction(moduleObj, name, argsObj);
+                if (callResult) {
+                    result = true;
+                    Py_DECREF(callResult);
+                }
+                Py_DECREF(argsObj);
             }
-
-            Py_DECREF(argsObj);
+            Py_DECREF(argObj);
```

Comment:
There is quite a lot of opportunity for simplification here. At minimum, you could make the argument object creation less tedious by doing `PyObject* argObj plPython::ConvertFrom(plPython::ToTuple, arg);`. Even better would be to offload all of this to the new Python vectorcall stuff, eg:
```C++
pyObjectRef moduleObj = ImportModule(module);
if (moduleObj) {
    pyObjectRef funcObj = PyObject_GetAttrString(moduleObj, name);
    if (funcObj)
        return (bool)plPython::CallObject(funcObj, args);
}
return false;
```
which will remove the `tp_call` translation penalty.

Reply:
Good idea - the code more straightforward with `plPython::CallObject`. (The performance benefits from vectorcall don't really matter here, because this API is used only by `Python.Cheat` and nothing else.)